extensions/gitpod-shared/src/analytics.ts

```diff
@@ -2,39 +2,75 @@
  *  Copyright (c) Gitpod. All rights reserved.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
+
+import { GitpodExtensionContext } from './features';
+
 export interface BaseGitpodAnalyticsEventPropeties {
-	sessionId: string
-	workspaceId: string
-	instanceId: string
-	appName: string
-	uiKind: 'web' | 'desktop'
-	devMode: boolean
-	version: string
-	timestamp: number
+	sessionId: string;
+	workspaceId: string;
+	instanceId: string;
+	appName: string;
+	uiKind: 'web' | 'desktop';
+	devMode: boolean;
+	version: string;
+	timestamp: number;
 }
 
-
 interface GAET<N extends string, P> {
-	eventName: N
-	properties: Omit<P, keyof BaseGitpodAnalyticsEventPropeties>
+	eventName: N;
+	properties: Omit<P, keyof BaseGitpodAnalyticsEventPropeties>;
 }
 
-
 export type GitpodAnalyticsEvent =
 	GAET<'vscode_session', {}> |
 	GAET<'vscode_execute_command_gitpod_open_link', {
-		url: string
+		url: string;
 	}> |
 	GAET<'vscode_execute_command_gitpod_change_vscode_type', {
-		targetUiKind: 'web' | 'desktop',
-		targetQualifier?: 'stable' | 'insiders'
+		targetUiKind: 'web' | 'desktop';
+		targetQualifier?: 'stable' | 'insiders';
 	}> |
 	GAET<'vscode_execute_command_gitpod_workspace', {
-		action: 'share' | 'stop-sharing' | 'stop' | 'snapshot' | 'extend-timeout'
+		action: 'share' | 'stop-sharing' | 'stop' | 'snapshot' | 'extend-timeout';
 	}> |
 	GAET<'vscode_execute_command_gitpod_ports', {
-		action: 'private' | 'public' | 'preview' | 'openBrowser'
+		action: 'private' | 'public' | 'preview' | 'openBrowser';
 	}> |
 	GAET<'vscode_execute_command_gitpod_config', {
-		action: 'remove' | 'add'
+		action: 'remove' | 'add';
+	}> |
+	GAET<'vscode_language_of_opened_file', {
+		lang: string; ext?: string;
 	}>;
+
+export function registerUsageAnalytics(context: GitpodExtensionContext): void {
+	context.fireAnalyticsEvent({ eventName: 'vscode_session', properties: {} });
+}
+
+const fileLanguageAccessMap: Record<string, boolean> = {};
+export function registerOpenedFileLanguageAnalytics(context: GitpodExtensionContext): void {
+	const track = (e: vscode.TextEditor) => {
+		const lang = e.document.languageId;
+		if (fileLanguageAccessMap[lang]) {
+			return;
+		}
+		fileLanguageAccessMap[lang] = true;
+		const ext = e.document.fileName.split('.').pop();
+		context.fireAnalyticsEvent({ eventName: 'vscode_language_of_opened_file', properties: { lang, ext } });
+	};
+
+	vscode.window.visibleTextEditors.forEach(e => {
+		if (e.document.fileName.toLowerCase().endsWith('readme.md')) {
```

Comment:
We don't need `README.md` in visibleTextEditors because it'll open if user did not specify one.

We need `other.<ext>` because when user open a workspace in GH with special file blob, this file will open as currentEditor

Reply:
Let's don't use visibleTextEditors but only activeTextEditor and without filtering, i.e.
```
track = () => {
   // get an active editor
   // if not tracked then track
}

track() 
vscode.window.onDidChangeActiveTextEditor(() => track());
```